src/algebra/ring/basic.lean

```diff
@@ -859,6 +859,31 @@ by rw [← mul_self_eq_mul_self_iff, one_mul]
 lemma units.inv_eq_self_iff (u : units α) : u⁻¹ = u ↔ u = 1 ∨ u = -1 :=
 by { rw inv_eq_iff_mul_eq_one, simp only [units.ext_iff], push_cast, exact mul_self_eq_one_iff }
 
+/--
+Makes a ring homomorphism from a additive group homomorphism from a commutative ring to an integral
+domain that commutes with self multiplication, assumes that two is nonzero and one is sent to one.
+-/
+def ring_hom.mk_mul_self_of_two_ne_zero [comm_ring β] (f : β →+ α) (h : ∀ x, f (x * x) = f x * f x)
```

Comment:
Should this be in the `add_hom` namespace (with a suitable name) so that dot notation can be used? If not, I think `f` can be made implicit.

Reply:
Yeah that's a nice idea, thanks! I moved the lemma too, but left them in the same file